test/es6/es6toLength.js

```diff
@@ -37,15 +37,15 @@ var tests = [
             assert.areEqual(-1,a.indexOf(3, 4294967290), "indexOf on an array is bound by array max length");
             assert.areEqual(-1,a.indexOf(4, 4294967290), "indexOf on an array is bound by array max length");
 
-            assert.areEqual(4294967294,Array.prototype.indexOf.call(o32, 2, 4294967290), "confirm we can find value 2 in the index range given");
-            assert.areEqual(4294967295,Array.prototype.indexOf.call(o32, 3, 4294967290), "objects don't have a bounded length so we should find the index given any length");
+            assert.areEqual(-1,Array.prototype.indexOf.call(o32, 2, 4294967290), "confirm we can find value 2 in the index range given");
+            assert.areEqual(-1,Array.prototype.indexOf.call(o32, 3, 4294967290), "objects don't have a bounded length so we should find the index given any length");

```

Comment:
Those unit test were not complaining with the specs. Here we are trying to get an item from an id > int32.Max. That is why we use o.length >>> 0; It is to shift o.length bits to the right, discarding bits shifted off, and shifting in zeroes from the left.